src/test/java/seedu/address/logic/commands/MarkPaidCommandTest.java

```diff
@@ -39,6 +40,47 @@ public void execute_markPaidPerson_success() {
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
     }
 
+    @Test
+    public void execute_duplicateMonthPaid_throwsCommandException() {
+        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        Index index = INDEX_FIRST_PERSON;
+        Set<MonthPaid> duplicateMonthsPaid = VALID_MONTHSPAID;
+
+
+        // Get the person and ensure they already have the months paid
+        Person person = model.getFilteredPersonList().get(index.getZeroBased());
+        Person personWithMonthsPaid = createMarkedPerson(person, duplicateMonthsPaid);
+        model.setPerson(person, personWithMonthsPaid);
+
+        // Attempt to add the same months again
+        MarkPaidCommand command = new MarkPaidCommand(index, duplicateMonthsPaid);
+
+        String duplicateMonth = duplicateMonthsPaid.iterator().next().toString(); // Assuming one month for simplicity
+        String expectedMessage = String.format("Duplicate month paid: %s", duplicateMonth);
+
+        assertCommandFailure(command, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_invalidMonthFormat_throwsCommandException() {
+        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        Index index = INDEX_FIRST_PERSON;
+
+        // Use an invalid month format, e.g., "2024-14"
+        MonthPaid invalidMonthPaid = new MonthPaid("[2024-14]");
```

Comment:
This shouldn't be possible: should be `new MonthPaid("2024-14")`

But looking at the code, was an oversight on my part to not add validation in MonthPaid.java. Will open an issue and add it now

Reply:
Also these tests should be moved to MarkpaidCommandParserTest